chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -144,16 +153,52 @@ impl actix::Actor for PeerManagerActor {
             }
         }));
 
+        // Periodically update the recent outbound connections in storage.
+        let clock = self.clock.clone();
+        let state = self.state.clone();
+        ctx.spawn(wrap_future(async move {
+            let mut interval =
+                time::Interval::new(clock.now(), UPDATE_RECENT_OUTBOUND_CONNECTIONS_INTERVAL);
+            loop {
+                interval.tick(&clock).await;
+                state.update_recent_outbound_connections(&clock);
+            }
+        }));
+
+        // Periodically make pending reconnect attempts.
+        let clock = self.clock.clone();
+        let state = self.state.clone();
+        ctx.spawn(wrap_future(async move {
+            let mut interval = time::Interval::new(clock.now(), RECONNECT_ATTEMPT_INTERVAL);
+            loop {
+                interval.tick(&clock).await;
+                for peer_info in state.get_peers_pending_reconnect() {
+                    let result = async {
+                        let stream = tcp::Stream::connect(&peer_info, tcp::Tier::T2).await.context("tcp::Stream::connect()")?;
+                        PeerActor::spawn_and_handshake(clock.clone(),stream,None,state.clone()).await.context("PeerActor::spawn()")?;
+                        anyhow::Ok(())
+                    }.await;
+
+                    if result.is_err() {
+                        tracing::info!(target:"network", ?result, "failed to connect to {peer_info}");
+                        if state.peer_store.peer_connection_attempt(&clock, &peer_info.id, result).is_err() {
+                            tracing::error!(target: "network", ?peer_info, "Failed to mark peer as failed.");
+                        }
+                    }
+                }
+            }
+        }));
+
         // Periodically prints bandwidth stats for each peer.
         self.report_bandwidth_stats_trigger(ctx, REPORT_BANDWIDTH_STATS_TRIGGER_INTERVAL);
     }
 
     /// Try to gracefully disconnect from connected peers.
     fn stopping(&mut self, _ctx: &mut Self::Context) -> actix::Running {
         tracing::warn!("PeerManager: stopping");
-        self.state
-            .tier2
-            .broadcast_message(Arc::new(PeerMessage::Disconnect(ClosingReason::Unknown)));
+        self.state.tier2.broadcast_message(Arc::new(PeerMessage::Disconnect(Disconnect {
+            allow_reconnect: true,
```

Comment:
document why allow_reconnect = true here

Reply:
I'm renaming this field to `remove_from_recent_outbound_connections` which should make it more clear how the value is used and why the default is false.